Add cropping and minSize to ImageComponent

Koerty opened the pull request:
This operations will be needed by the grid later on the road so I better submit the changes now in its own pull request.

This add new methods to the class ImageComponent:
```c++
void setMinSize(float width, float height);
void setMinSize(const Vector2f& size);
void cropLeft(float percent);
void cropTop(float percent);
void cropRight(float percent);
void cropBot(float percent);
void crop(float left, float top, float right, float bot);
void uncrop();
```

Usage examples : 
- A call to cropTop(0.2) will crop 20% of the top of the image
- A call to minSize(0.2, 0.2) will resize the image smallest side to 0.2 without breaking aspect ratio, but crop the extra lenght of the biggest side.
- Call minSize(0.2, 0.2) and then uncrop() to define an image size by its smallest side

jrassa commented:
I am getting compile errors trying to build this.  Looks like mTargetIsMin is missing from ImageComponent.h

jrassa commented:
I added it in so I could test.  Looks good to me.  Once you get that fixed I can merge it.

Koerty commented:
Fixed and rebased.

To create this PR I just cherrypicked 2 commits from one of my experimental branch and squashed them. I was so sure it would work I didn't even bother trying it. Sorry this isn't very professional, I should have performed more testing.

jrassa commented:
Thanks.
